sdk/cosmosdb/cosmos/src/queryExecutionContext/QueryRangeMapping.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { PartitionKeyRange } from "../index.js";
+
+/**
+ * @hidden
+ * Extended partition key range that includes effective partition key (EPK) boundaries
+ * for handling partition split and merge scenarios
+ */
+export interface ExtendedPartitionKeyRange extends PartitionKeyRange {
+  /**
+   * Effective partition key minimum boundary (used for split/merge operations)
+   */
+  epkMin?: string;
+
+  /**
+   * Effective partition key maximum boundary (used for split/merge operations)
+   */
+  epkMax?: string;
+}
+
+/**
+ * @hidden
+ * Represents a range mapping for query execution context
+ */
+export interface QueryRangeMapping {
+  /**
+   * Number of items from this partition range in the current buffer
+   */
+  itemCount: number;
+
+  /**
+   * Continuation token for this partition key range
+   */
+  continuationToken: string | null;
+
+  /**
+   * The extended partition key range this mapping belongs to (includes EPK boundaries)
+   */
+  partitionKeyRange?: ExtendedPartitionKeyRange;
+
+  /**
+   * Hash of the last document result for this partition key range (for distinct queries)
+   */
+  hashedLastResult?: string;
+
+  offset?: number;
+
+  limit?: number;
+}
+
+/**
+ * @hidden
+ * Creates an ExtendedPartitionKeyRange from a regular PartitionKeyRange
+ * @param partitionKeyRange - The base partition key range
+ * @param epkMin - Optional effective partition key minimum boundary
+ * @param epkMax - Optional effective partition key maximum boundary
+ * @returns Extended partition key range with EPK boundaries
+ */
+export function createExtendedPartitionKeyRange(
```

Comment:
These helper functions could be moved to a new file queryUtils

Reply:
removed